components/Modal.tsx

```diff
@@ -0,0 +1,80 @@
+import { Dialog, Transition } from "@headlessui/react";
+import { forwardRef, Fragment, useImperativeHandle, useState } from "react";
+import { XMarkIcon } from "@heroicons/react/20/solid";
+
+interface Modal {
+  open: () => void;
+  close: () => void;
+}
+
+interface Props {
+  title?: string;
+  children: React.ReactNode;
+}
+
+const Modal = forwardRef<Modal, Props>(({ title, children }, ref) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  function open() {
+    setIsOpen(true);
+  }
+
+  function close() {
+    setIsOpen(false);
+  }
+
+  useImperativeHandle(ref, () => ({
+    open,
+    close,
+  }));
+
+  return (
+    <Transition.Root show={isOpen} as={Fragment}>
+      <Dialog as="div" className="relative z-10" onClose={close}>
+        <Transition.Child
+          as={Fragment}
+          enter="ease-out duration-300"
+          enterFrom="opacity-0"
+          enterTo="opacity-100"
+          leave="ease-in duration-200"
+          leaveFrom="opacity-100"
+          leaveTo="opacity-0"
+        >
+          <div className="fixed inset-0 bg-black/40 backdrop-blur-lg transition-opacity" />
+        </Transition.Child>
+
+        <div className="fixed inset-0 z-10 overflow-y-auto ">
+          <div className="flex min-h-full items-end justify-center p-4 text-center sm:items-center sm:p-0">
+            <Transition.Child
+              as={Fragment}
+              enter="ease-out duration-300"
+              enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+              enterTo="opacity-100 translate-y-0 sm:scale-100"
+              leave="ease-in duration-200"
+              leaveFrom="opacity-100 translate-y-0 sm:scale-100"
+              leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+            >
+              <Dialog.Panel className="relative transform overflow-hidden rounded-lg bg-gray-200 px-4 pt-5 pb-4 text-left shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-lg sm:p-6">
+                {typeof title === "string" && (
```

Comment:
create a typeguard for this something like:

`const isString = (value: unkown): value is string => typeof value === 'string' `

Reply:
Is this necessary if it's only being used once here, I can understand if there was repeated use.

Reply:
I think as this thing grows it would be useful to have some utility functions, especially if you might want to use this as a project to show someone but can keep it as is for now if you like.